### ISSUE TITLE
[Consumption][Transversal] To characterize the consumption of the device

### ISSUE THREAD

**Issue by ralcaide** [Sinapse-Energia/talking-fiber#5]
After testing the MVP we saw the consumption is very high. It is necessary to characterize this consumption. This is a transversal task of the next milestones.

Consumption to be measured:

- Current MVP publishing each 1m, 5m, 15m and 30m
- All peripherals OFF and MCU ON
- All peripherals OFF and MCU in stand by
- Quectel connecting
- Quectel publishing / receiving
- Others...

**Comment by ghost** [comment]
We have found that consumption of photodiode measurement part is about 275-300 mA. It is a lot. In MVP code we enabled photodiode at start and not disabled it at all. After last commit photodiode will be enabled only when measurement performed, it should extend a lot battery life.

Please check the schematic, usually such measurement should not consume so much current.

**Comment by ralcaide** [comment]
Yes, this is a lot. Thanks. We will check the schematics and test the new code

**Comment by ralcaide** [comment]
Testing in progress. Battery used JH 903052P 3.7V 1200mAh

Publishing each minute: 11 Hours
Publishing each 15 minutes: 18 Hours (Still ON)
Publishing each 30 minutes: TBD

With the photo-diode disconnection can be observed an improvement between 1m vs 15m (before last commit this change was not observed)

Questions:

- The device is publishing a debug message each 40 seconds. Is this fact consuming battery?
- The device is reconnecting several times during the hour because bad coverage. Is this fact consuming battery?

Please, characterize the consumption of the quectel M95, we suppose the connection process consumes a lot of battery

**Comment by ralcaide** [comment]
Update:

Publishing each 15 min: 20 Hours

The battery life is still very low. It is necessary to understand where is going the consumption and improve it

**Comment by ralcaide** [comment]
Ivan says:

> We have ammeter built in power suply + external lab multimeter. when measurement circuit enabled consumption is 350 mA on 3.7V, when measurement disabled - consumption is about 60 mA

If it is like that, a battery with 1200mAh, if the photodiode is disabled, will have a maximum lifetime of 20h. Right? This is the result already archived with FW publishing each 15 minutes. 

This lifetime is very low, we need to improve it. I suppose after #6 and #7 the results will improve a lot

**Comment by ghost** [comment]
FYI: When analyzing schematic we found several issues that should be fixed to have better power performance and stability of the board:
1) Seems like 3.7V is directed to the VBAT input of MCU, please check this, it may cause problems in future.
2) We have no external crystal for RTC (32768Hz), this is limit us in usage for some advanced power save modes.
3) In some sleep modes with low power consumption GPIOs are disabled. Seems like when EX_RESET_PHOTODIODE MCU pin is disabled, photodiode circuit is switched on, maybe because R_17_10 is too large. We will change it from 360K to 10K and test.

**Comment by ghost** [comment]
Please also check the board with multimeter/scope. On our board when EX_RESET_PHOTODIODE is disabled we still have 1V at the output of IC_67_2.

**Comment by ralcaide** [comment]
Thank you for the comments. We will re-check the design. Let us know please the results of your tests

**Comment by ralcaide** [comment]
Regarding 2. What limitations have you found? What modes are we not able to use?

**Comment by ghost** [comment]
Shutdown mode is required LSE to be present to allow RTC ticks. But I think we will mostly use Standby mode. In this mode main clocks is disabled, but we will check if LSI RTC clock is working.

**Comment by ralcaide** [comment]
Ok, check please if the Standby mode is possible with LSI and let us know. If it is not possible let us know asap in order to check the design and maybe add an LSE

**Comment by ghost** [comment]
While Eugene analyzed the board he found fix that we can use in order to avoid high consumption on sleep, please test the latest commit

**Comment by ghost** [comment]
There is also another issue - now we are using amplifier in always shutdown mode, because in active mode its output signal is always 12 mV or some random value around 500-1000 mV. Eugene is investigating this issue now.

**Comment by ralcaide** [comment]
@fpgamcu , what is the status of this task? have Eugene investigated about that?

**Comment by ghost** [comment]
Yes, we had damaged IC_67_2 on our board, after I resoldered it consumption and measurements now is ok.
